interfaces/apparmor/backend.go

```diff
@@ -103,7 +103,26 @@ func (b *Backend) Setup(snapInfo *snap.Info, developerMode bool, repo *interface
 	if errReload != nil {
 		return errReload
 	}
-	return errUnload
+	if errUnload != nil {
+		return errUnload
+	}
+	// Sanity checking, see if all the profiles are really loaded and load
+	// those profiles that are not. This can happen if the files were on disk
+	// but were not really loaded (e.g. due to an interrupted install).
```

Comment:
No, this is not actually what the patched code is doing. The code does a complex dance trying to observe what actually changed, and then makes it even more complex by observing if any of the files on disk are not loaded.

What we should do instead, is to load everything the change _might_ possibly have changed. I suspect that's going to be much simpler, more robust, and perhaps even faster, ironically, since we won't be re-reading stuff from disk.